app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,36 @@
+class CommentsController < ApplicationController
+
+  def new
+    if params[:answer_id]
+      @commentable = Answer.find(params[:answer_id])
+    else
+      @commentable = Question.find(params[:question_id])
+    end
+    @comment = Comment.new
+  end
+
+
+  def create
+    if params[:answer_id]
+      @commentable = Answer.find(params[:answer_id])
+    else
+      @commentable = Question.find(params[:question_id])
+    end
+
```

Comment:
This is used a couple of times - it might be worth implementing a private "load_commentable" that returns the right commentable.
Then here could just be `@commentable = find_commentable`